### PR TITLE
Reduce gas usage for opening loans

### PR DESCRIPTION
**Pull request by barrasso**
The goal of this PR is to reduce the gas used when calling `CollateralShort.open` so it doesn't hit the limit on L2.

This requires a balance between the contract size of `CollateralShort` and the gas usage of `_open`. 

Ideally, we want to minimize any external contract calls since they are quite costly.

### CODE REVIEW

File: contracts/Collateral.sol [hunk 1]
```diff
@@ -251,7 +271,7 @@ contract Collateral is ICollateralLoan, Owned, MixinSystemSettings {
         require(amount <= maxLoan(collateral, currency), "Exceed max borrow power");
 
         // 7. This fee is denominated in the currency of the loan.
-        uint issueFee = amount.multiplyDecimalRound(getIssueFeeRate(address(this)));
+        uint issueFee = amount.multiplyDecimalRound(issueFeeRate);
```
Comment (barrasso) [inline contracts/Collateral.sol:274]: I am seeing a reduction of ~1M *(!)* gas used by moving `issueFeeRate` back into the `Collateral` contract.

File: test/contracts/Collateral.js [hunk 2]
```diff
@@ -15,7 +15,13 @@ contract('Collateral', async accounts => {
 		ensureOnlyExpectedMutativeFunctions({
 			abi: Collateral.abi,
 			ignoreParents: ['Owned', 'Pausable', 'MixinResolver', 'Proxy'],
-			expected: ['addRewardsContracts', 'addSynths'],
+			expected: [
+				'addRewardsContracts',
```
Comment (justinjmoses) [inline test/contracts/Collateral.js:19]: at the very least please add tests to make sure they are `onlyOwner` using `onlyGivenAddressCanInvoke`

File: test/contracts/CollateralShort.js [hunk 3]
```diff
@@ -219,6 +230,70 @@ contract('CollateralShort', async accounts => {
...
+				beforeEach(async () => {
+					await short.setCanOpenLoans(false, { from: owner });
+				});
+				it('should update the flag', async () => {
+					assert.isFalse(await short.canOpenLoans());
+				});
+			});
+		});
+
+		describe('setMinCollateral', async () => {
```
Comment (justinjmoses) [inline test/contracts/CollateralShort.js:253]: these should be moved to the `Collateral` tests.

Comment (justinjmoses) [reply test/contracts/CollateralShort.js:253]: Also look at using the `onlyGivenAddressCanInvoke` function to test `onlyOwner`

File: contracts/Collateral.sol [hunk 4]
```diff
@@ -173,6 +176,23 @@ contract Collateral is ICollateralLoan, Owned, MixinSystemSettings {
         return true;
     }
 
+    /* ---------- SETTERS ---------- */
+
+    function setMinCollateral(uint _minCollateral) external onlyOwner {
```
Comment (dbeal-eth) [inline contracts/Collateral.sol:181]: why were these setters missing before, if they were never removed from the `Collateral` contract?

File: test/contracts/Collateral.js [hunk 5]
```diff
@@ -15,7 +15,13 @@ contract('Collateral', async accounts => {
...
+				'addSynths',
+				'setCanOpenLoans',
+				'setIssueFeeRate',
+				'setMinCollateral',
+			],
 		});
```
Comment (dbeal-eth) [inline test/contracts/Collateral.js:25]: do tests need to be added to this file to verify the correct funcitoning of the `set` functions?

File: test/contracts/SystemSettings.js [hunk 6]
```diff
@@ -546,6 +506,9 @@ contract('SystemSettings', async accounts => {
 	});
 
 	describe('setLiquidationRatio()', () => {
+		before(async () => {
+			await setupSettings();
```
Comment (dbeal-eth) [inline test/contracts/SystemSettings.js:510]: why is this added here?

File: test/integration/behaviors/short.behavior.js [hunk 7]
```diff
@@ -90,7 +88,10 @@ function itCanOpenAndCloseShort({ ctx }) {
 				before('open the loan', async () => {
 					tx = await CollateralShort.open(amountToDeposit, amountToBorrow, toBytes32('sETH'));
 
-					const { events } = await tx.wait();
+					const receipt = await tx.wait();
+					console.log(`short.open gas used: ${receipt.gasUsed.toString()}`);
```
Comment (dbeal-eth) [inline test/integration/behaviors/short.behavior.js:92]: remove log

**Review by dbeal-eth — APPROVED** [review]
look before merging
